definitions/checks/puppet/provide_upgrade_guide.rb

```diff
@@ -0,0 +1,36 @@
+module Checks::Puppet
+  class ProvideUpgradeGuide < ForemanMaintain::Check
+    metadata do
+      description 'Verify puppet and provide upgrade guide for it'
+      tags :puppet_upgrade_guide
+      confine do
+        feature(:downstream) &&
+          feature(:downstream).current_minor_version == '6.3' &&
+          find_package('puppet')
+      end
+      manual_detection
+    end
+
+    def run
+      with_spinner('Verifying puppet version before upgrade') do |spinner|
+        puppet_package = find_package('puppet')
+        spinner.update "current puppet version: #{puppet_package}"
+        result = (puppet_package =~ /puppet-3/)
+        curr_sat_version = feature(:downstream).current_minor_version
+        assert(
+          result.nil?,
```

Comment:
It would read a bit better to not use the result variable, but use the condition directly here

```ruby
assert(
  (puppet_package !~ /puppet-3/),
  …
```

Reply:
Done!